src/kubernetes.h

```diff
@@ -141,6 +165,17 @@ class KubernetesReader {
   json::value FindTopLevelController(const std::string& ns, json::value object)
       const throw(QueryException, json::Exception);
 
+  // Update service_to_metadata_ cache based on a newly updated service.
+  void UpdateServiceToMetadataCache(
+      const json::Object* service, bool is_deleted) throw(json::Exception);
+
+  // Update service_to_pods_ cache based on a newly updated endpoints.
+  void UpdateServiceToPodsCache(
+      const json::Object* endpoints, bool is_deleted) throw(json::Exception);
+
+  // Const data.
```

Comment:
This isn't helpful. How about `// An empty vector value for endpoints that have no pods.`?